jenkins/deploy/Jenkinsfile

```diff
@@ -0,0 +1,96 @@
+pipeline {
+    agent any
+
+    options {
+        buildDiscarder(logRotator(numToKeepStr: "5"))
+        timeout time: 60, unit: "MINUTES"
+    }
+
+    stages {
+        stage("Build") {
+            steps {
+                configFileProvider([configFile(fileId: "iis-build-properties", variable: 'BUILD_PROPERTIES')]) {
+                    load "${BUILD_PROPERTIES}"
+                    withEnv(["JAVA_HOME=${ tool type: 'jdk', name: "${JDK_VERSION}" }",
+                             "PATH+MAVEN=${tool type: 'maven', name: "${MAVEN_VERSION}"}/bin:${env.JAVA_HOME}/bin"]) {
+                        sh """
+                            mvn clean compile
+                        """
+                    }
+                }
+            }
+        }
+
+        stage("Test") {
+            steps {
+                configFileProvider([configFile(fileId: "iis-build-properties", variable: 'BUILD_PROPERTIES')]) {
+                    load "${BUILD_PROPERTIES}"
+                    withEnv(["JAVA_HOME=${ tool type: 'jdk', name: "$JDK_VERSION" }",
+                             "PATH+MAVEN=${tool type: 'maven', name: "$MAVEN_VERSION"}/bin:${env.JAVA_HOME}/bin"]) {
+                        sh """
+                            mvn test
+                        """
+                    }
+                }
+            }
+        }
+
+        stage("Install") {
```

Comment:
`Test` and `Install` were decoupled because I wanted to run unit tests and not run integration tests when deploying. In an optimal solution we probably should run all the tests but because integration tests take vast amount of time we cannot run them during deployment process. On the other hand unit tests take only a few minutes and can be run during deployment, this will also protect us from deploying failing code i.e. when deploying from a branch different then `master`. 

`Build` and `Test` distinction is not necessary, it was probably taken from some example Jenkins setup found on the net. I've removed `Build` step from all Jenkinsfiles.

Reply:
Let's not forget about the beautiful Stage View for the jobs in Jenkins. How sad would it look if we only had one stage? :-)